File: examples-sdk/typescript/301-rag-basic-vector-search.ts
#!/usr/bin/env tsx
// 301-rag-basic-vector-search — TypeScript SDK equivalent
// Compile: vil compile --from typescript --input 301-rag-basic-vector-search.ts --release

import { VilPipeline, VilServer, ServiceProcess } from "vil-sdk";

const server = new VilServer("rag-basic-vector-search", 3110);
const rag_basic = new ServiceProcess("rag-basic");
rag_basic.endpoint("POST", "/rag", "rag_handler");
server.service(rag_basic);
server.compile();

File: examples-sdk/python/301-rag-basic-vector-search.py
#!/usr/bin/env python3
"""301-rag-basic-vector-search — Python SDK equivalent
Compile: vil compile --from python --input 301-rag-basic-vector-search.py --release
"""
import os
from vil import VilPipeline, VilServer, ServiceProcess

server = VilServer("rag-basic-vector-search", port=3110)
rag_basic = server.service_process("rag-basic")
rag_basic.endpoint("POST", "/rag", "rag_handler")
server.compile()
